app/components/Analyst/CBC/CbcEditProjectNumber.tsx

```diff
@@ -0,0 +1,229 @@
+  display: inline-block;
+  margin-top: 8px;
+  flex-basis: 100%;
+`;
+
+const StyledMessage = styled('div')`
+  display: flex;
+  &::after {
+    content: '.';
+    visibility: hidden;
+  }
+  flex-basis: 100%;
+`;
+
+const CbcEditProjectNumber: React.FC<Props> = ({
+  cbc,
+  value,
+  isHeaderEditable,
+}) => {
+  const queryFragment = useFragment(
+    graphql`
+      fragment CbcEditProjectNumber_query on Cbc {
+        projectNumber
+      }
+    `,
+    cbc
+  );
+
+  const [baseProjectNumber, setBaseProjectNumber] = useState(value);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isValidating, setIsValidating] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
+  const [updateProjectNumber] = useUpdateCbcProjectNumberMutation();
+
+  const [projectNumber, setProjectNumber] = useState(
+    queryFragment?.projectNumber
+  );
+
+  const [error, setError] = useState(null);
+
+  const handleSubmit = () => {
+    setIsSaving(true);
+    const parsedValue = parseInt(projectNumber, 10);
+    if (baseProjectNumber !== parsedValue) {
+      updateProjectNumber({
+        variables: {
+          input: {
+            projectNumber: baseProjectNumber,
+            cbcPatch: {
+              projectNumber: parsedValue,
+            },
+          },
+        },
+        onCompleted: () => {
+          setProjectNumber(parsedValue);
+          setBaseProjectNumber(parsedValue);
+          setIsEditing(false);
+        },
+        onError: (e) => {
+          if (e?.message?.indexOf('duplicate key') !== -1) {
+            setError('A project with this number already exists.');
+          } else {
+            setError(
+              'An error occurred while attempting to update the project number.'
+            );
+          }
+          setIsSaving(false);
+          setIsEditing(false);
+          setProjectNumber(baseProjectNumber);
+        },
+        debounceKey: 'cbc_change_project_number',
+      });
+    } else {
+      setIsSaving(false);
+      setIsEditing(false);
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      if (!error) handleSubmit();
+    } else if (e.key === 'Escape') {
+      setIsEditing(false);
+      setProjectNumber(value);
+    }
+  };
+
+  const checkProjectNumberIsUnique = async (val) => {
+    try {
+      const response = await fetch('/api/validation/project-number-unique', {
+        method: 'POST',
+        body: JSON.stringify({ projectNumber: val }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+      if (!response.ok) return false;
+      const data = await response.json();
+      return data;
+    } catch (e) {
+      setError(
+        'An error occurred while attempting to validate the project number.'
+      );
+      return false;
+    }
+  };
+
+  const validateProjectNumber = async (val) => {
+    setIsValidating(true);
```

Comment:
I think we need to validate if the project number is <4 as well?

> When editing and on read only display If “Project Number” is not 4 digits integer, validate with red box outline and text.  “Invalid number, please use a 4 digit number”.

Reply:
I thought so too at first, but AC states "Allow user to save the the project number with >=1 digit.", so current validation should be ok.